app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/meta/FilterUtil.java

```diff
@@ -0,0 +1,91 @@
+package io.syndesis.connector.mongo.meta;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class FilterUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FilterUtil.class);
+    private static final Pattern PARAM_PATTERN = Pattern.compile(":#([^ \\p{Punct}]+)");
+
+    private FilterUtil() {
+
+    }
+
+    /**
+     * Extract any <strong>:#variable</strong> parameter format
+     *
+     * @param filter the filter expression, ie {"test": ":#var1"}
+     * @return a list of parameters expected by the filter
+     */
+    public static List<String> extractParameters(String filter) {
+        List<String> result = new ArrayList<>();
+        Matcher matcher = PARAM_PATTERN.matcher(filter);
+        while (matcher.find()) {
+            String param = matcher.group();
+            result.add(param.substring(2));
+        }
+        return result;
+    }
+
+    /**
+     * Check whether the filter contains any parameter or is a static one
+     *
+     * @param filter
+     * @return
+     */
+    public static boolean hasAnyParameter(String filter) {
+        return filter != null && filter.contains(":#");
+    }
+
+    /**
+     * Merge the parameters found in the filter with the values found in the JSON message
+     *
+     * @param filter an expression containing :#variable parameters
+     * @param jsonMessage a message formatted as JSON
+     * @return
+     */
+    public static String merge(String filter, String jsonMessage) {
+        LOGGER.debug("Filter configured by user {}", filter);
+        LOGGER.debug("Input body {}", jsonMessage);
+        String mergedFilter = filter;
+        if (mergedFilter != null) {
```

Comment:
I'd prefer early exit from method like:
```suggestion
        if (mergedFilter == null) {
        	LOGGER.debug("Skipping merge as filter was null");
        	return null;
        }
```